### DISCUSSION TITLE
Can `forge test` send txs to a real network?

### DISCUSSION THREAD

**Discussion by mtomassoli** [foundry-rs/foundry#4217] [Q&A] [answered]
Is it possible for `forge test` to send a tx to a real network?
In other words, is there any combination of env variables, CLI parameters, and test code such that `forge test [+ CLI parameters]` sends a transaction to a real network?

**Comment by mds1** [comment, marked as answer]
`forge test` cannot send transactions to a network, however that is what `forge script` is intended to do

**Reply by mtomassoli** [reply]
I know about `forge script`. I'm asking for when I'll be forking a network and test really dangerous POCs with `forge test`. To be extra sure, I'll probably use a sandbox where I'll never use any valid private key. I want to drive the probability of things going wrong to 0.

**Reply by mds1** [reply]
`forge test` can send RPC requests to fetch data, e.g. if you're running tests against a forked network, but it cannot send signed transactions to the network (and you don't need to give it a private key)

**Reply by mtomassoli** [reply]
One last question. I noticed that `broadcast`, `startBroadcast`, and `endBroadcast` work even in tests. Are they no-ops or do they collect txs just like in the first phase of `forge script`?

**Reply by mds1** [reply]
When invoked during tests I believe they just act like `vm.prank` would

**Reply by mtomassoli** [reply]
I completely forgot about `ffi`. So `forge test` can indeed send txs *indirectly*! Again: sandbox.
